sale_order_lock_stock_picking/models/stock_picking.py

```diff
@@ -0,0 +1,29 @@
+# © 2021 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3.0 (https://www.gnu.org/licenses/lgpl-3.0.html)
+
+from odoo import models, api, _
+from odoo.exceptions import ValidationError
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    def lock_validation_error(self):
+        if self.sale_id.state == 'done':
+            raise ValidationError( _(
+                "The related sales order is locked. Unlock to continue."
+                )
+            )
+
+    def button_validate(self):
+        self.lock_validation_error()
+        super(StockPicking, self).button_validate()        
+
+    def write(self, vals):
+        self.lock_validation_error()
+        return super(StockPicking, self).write(vals)
+
+    @api.model
+    def create(self, vals):
+        self.lock_validation_error()
```

Comment:
This won't work IMO, because `self` has no values, but `vals` does. So `vals['sale_id']` should be inspected inst4ead.